File: android/wsvideoeditor-test/src/main/java/com/whensunset/wsvideoeditortest/AppImpl.java
package com.whensunset.wsvideoeditortest;

import android.app.Application;
import android.content.res.AssetManager;
import android.util.Log;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

/**
 * 2020/8/10
 *
 * @author zhengliao
 */
public class AppImpl extends Application {


    @Override
    public void onCreate() {
        super.onCreate();
        copyAssets();
    }

    private void copyAssets() {
        AssetManager assetManager = getAssets();
        InputStream in = null;
        OutputStream out = null;
        try {
            in = assetManager.open("test.mp4");
            File outFile = new File(getExternalFilesDir(null), "test.mp4");
            out = new FileOutputStream(outFile);
            copyFile(in, out);
        } catch (IOException e) {
            Log.e("tag", "Failed to copy asset file: " + "test.mp4", e);
        } finally {
            if (in != null) {
                try {
                    in.close();
                } catch (IOException e) {
                    // NOOP
                }
            }
            if (out != null) {
                try {
                    out.close();
                } catch (IOException e) {
                    // NOOP
                }
            }
        }
    }

    private void copyFile(InputStream in, OutputStream out) throws IOException {
        byte[] buffer = new byte[1024];
        int read;
        while ((read = in.read(buffer)) != -1) {
            out.write(buffer, 0, read);
        }
    }
}

File: android/wsvideoeditor-test/src/main/java/com/whensunset/wsvideoeditortest/AppImpl.kt
package com.whensunset.wsvideoeditortest

import android.app.Application
import android.util.Log
import java.io.*
import kotlin.jvm.Throws

/**
 * 2020/8/10
 *
 * @author zhengliao
 */

class AppImpl : Application() {
    override fun onCreate() {
        super.onCreate()
        copyAssets()
    }

    private fun copyAssets() {
        val assetManager = assets
        var `in`: InputStream? = null
        var out: OutputStream? = null
        try {
            `in` = assetManager.open("test.mp4")
            val outFile = File(getExternalFilesDir(null), "test.mp4")
            out = FileOutputStream(outFile)
            copyFile(`in`, out)
        } catch (e: IOException) {
            Log.e("tag", "Failed to copy asset file: " + "test.mp4", e)
        } finally {
            if (`in` != null) {
                try {
                    `in`.close()
                } catch (e: IOException) {
                    // NOOP
                }
            }
            if (out != null) {
                try {
                    out.close()
                } catch (e: IOException) {
                    // NOOP
                }
            }
        }
    }

    @Throws(IOException::class)
    private fun copyFile(`in`: InputStream?, out: OutputStream) {
        val buffer = ByteArray(1024)
        var read: Int
        while (`in`!!.read(buffer).also { read = it } != -1) {
            out.write(buffer, 0, read)
        }
    }
}